schema/schema_test.go

```diff
@@ -0,0 +1,295 @@
+		Version:         version,
+		ColumnSchema:    columnSchema,
+		ParentMigration: migration,
+	}
+}
+
+func buildColumnDefinition(inboundName, outboundName, transformer, columnCreationOptions string) ColumnDefinition {
+	return ColumnDefinition{
+		InboundName:           inboundName,
+		OutboundName:          outboundName,
+		Transformer:           transformer,
+		ColumnCreationOptions: columnCreationOptions,
+	}
+}
+
+func buildColumnDefinitionBatch(num int, inboundName, outboundName, transformer, columnCreationOptions string) ColumnDefinition {
+	return ColumnDefinition{
+		InboundName:           fmt.Sprintf(inboundName+"%d", num),
+		OutboundName:          fmt.Sprintf(outboundName+"%d", num),
+		Transformer:           transformer,
+		ColumnCreationOptions: columnCreationOptions,
+	}
+}
+
+func buildColumns(start, end int, inboundName, outboundName, transformer, columnCreationOptions string) []ColumnDefinition {
+	var columns []ColumnDefinition
+	for i := start; i < end; i++ {
+		columns = append(columns, buildColumnDefinitionBatch(i, inboundName, outboundName, transformer, columnCreationOptions))
+	}
+	return columns
+}
+
+func buildColumnOperationBatch(num int, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions string) ColumnOperation {
+	return ColumnOperation{
+		Operation:    operation,
+		InboundName:  fmt.Sprintf(inboundNameKey+"%d", num),
+		OutboundName: fmt.Sprintf(outboundNameKey+"%d", num),
+		NewColumnDefinition: ColumnDefinition{
+			InboundName:           fmt.Sprintf(inboundName+"%d", num),
+			OutboundName:          fmt.Sprintf(outboundName+"%d", num),
+			Transformer:           transformer,
+			ColumnCreationOptions: columnCreationOptions,
+		},
+	}
+}
+
+func buildColumnOperations(start, end int, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions string) []ColumnOperation {
+	var columnOperations []ColumnOperation
+	for i := start; i < end; i++ {
+		columnOperations = append(columnOperations, buildColumnOperationBatch(i, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions))
+	}
+	return columnOperations
+}
+
+func buildTableOption(distKey, sortKey []string) TableOption {
+	return TableOption{
+		DistKey: distKey,
+		SortKey: sortKey,
+	}
+}
+
+func buildStringList(list ...string) []string {
+	if list == nil {
+		return []string{}
+	}
+	return list
+}
+
+func buildColumnSchema(tableOption TableOption, columns []ColumnDefinition) ColumnSchema {
+	return ColumnSchema{
+		TableOption: tableOption,
+		Columns:     columns,
+	}
+}
+
+func buildMigration(tableOperation, name string, columnOperations []ColumnOperation, tableOption TableOption) Migration {
+	return Migration{
+		TableOperation:   tableOperation,
+		Name:             name,
+		ColumnOperations: columnOperations,
+		TableOption:      tableOption,
+	}
+}
+
+func TestAddTable(t *testing.T) {
```

Comment:
Consider factoring out your boilerplate and running these as separate tests so that if multiple fail they don't prevent each other from telling you what's wrong.